diam/server.go

```diff
@@ -137,6 +142,45 @@ func (c *conn) notifyClientGone() {
 	}
 }
 
+// A ConnState represents the state of a client connection to a server.
+type ConnState int
```

Comment:
why not int8 here?

Reply:
taken from https://github.com/golang/go/blob/master/src/net/http/server.go#L2823

looks like it can be an int8 afaict, can update